webapp/src/main/java/com/box/l10n/mojito/security/RequestLoggingFilter.java

```diff
@@ -0,0 +1,36 @@
+package com.box.l10n.mojito.security;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class RequestLoggingFilter extends OncePerRequestFilter {
+
+  private static final Logger logger = LoggerFactory.getLogger(RequestLoggingFilter.class);
+
+  private final RequestLoggingConfig requestLoggingConfig;
+  private final RequestLoggerHelper requestLoggerHelper;
+
+  public RequestLoggingFilter(
+      RequestLoggerHelper requestLoggerHelper, RequestLoggingConfig requestLoggingConfig) {
+    this.requestLoggingConfig = requestLoggingConfig;
+    this.requestLoggerHelper = requestLoggerHelper;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws jakarta.servlet.ServletException, IOException {
+
+    if (requestLoggingConfig.isLoggingEnabled() && requestLoggerHelper != null) {
+      String msg = requestLoggerHelper.summarizeRequest(request, requestLoggingConfig);
```

Comment:
Should we add filtering for specific api paths e.g. only log requests to the /login endpoint?

Reply:
Unfortunately that's not applicable when you log into mojito-dev. If you are using oauth, it doesn't even call the /login endpoint. It calls various oauth endpoints

The final desired flow the /login endpoint might not even be used except for very rare cases. So going forward, we might want to enable logging more generally too.